mmcv/runner/hooks/checkpoint.py

```diff
@@ -32,59 +35,57 @@ def __init__(self,
                  save_optimizer=True,
                  out_dir=None,
                  max_keep_ckpts=-1,
+                 sync_buffer=True,
                  **kwargs):
         self.interval = interval
         self.by_epoch = by_epoch
         self.save_optimizer = save_optimizer
         self.out_dir = out_dir
         self.max_keep_ckpts = max_keep_ckpts
         self.args = kwargs
+        self.sync_buffer = sync_buffer
+        self.sync_buffer_hook = SyncBuffersHook()
 
-    @master_only
     def after_train_epoch(self, runner):
         if not self.by_epoch or not self.every_n_epochs(runner, self.interval):
             return
 
         runner.logger.info(f'Saving checkpoint at {runner.epoch + 1} epochs')
+        if self.sync_buffer:
+            self.sync_buffer_hook.sync_buffers(runner)
```

Comment:
I suggest adding a utility method `sync_buffer(model)` for both hooks, instead of using one hook inside another.